### ISSUE TITLE
Cannot build kotlin project with --no-daemon option

### ISSUE THREAD

**Issue by z0mb1ek** [gradle/gradle#17611]
Hi, I use spring-boot with kotlin and want to make boot jar.
But then i try to build it in some docker jdk image with command `./gradlew bootJar --no-daemon` i get:
```
FAILURE: Build failed with an exception.
 
* What went wrong:
java.lang.ExceptionInInitializerError (no error message)
> Unable to make protected void java.util.ResourceBundle.setParent(java.util.ResourceBundle) accessible: module java.base does not "opens java.util" to unnamed module @1caa0244
```

Without `--no-daemon` it work well. I use `--no-daemon` for my java projects and it works well too. Only with kotlin there are troubles.

Gradle 7.1.1
Kotlin 1.5.20
java 16

**Comment by blindpirate** [comment]
Can you share a minimal reproducible project? Without reproducer we may not able to help.

**Comment by mlopatkin** [comment]
Thank you for your interest in Gradle!

This issue will be closed as a duplicate of #17229. Please add your +1's to that issue.

Please check https://github.com/gradle/gradle/issues/17229#issuecomment-863232813 for the explanation and possible workarounds.
